tool/gravity/cli/clusterupdate.go

```diff
@@ -428,6 +431,29 @@ func (r *clusterInitializer) validatePreconditions(localEnv *localenv.LocalEnvir
 	return nil
 }
 
+// checkAppIntegrity makes sure that all upgrade application dependencies
+// are present in the cluster store.
+func (r *clusterInitializer) checkAppIntegrity(env *localenv.LocalEnvironment, upgradeApp *app.Application) error {
+	packages, err := env.ClusterPackages()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	apps, err := env.AppServiceCluster()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	err = app.VerifyDependencies(upgradeApp, apps, packages)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to verify %v dependencies.", upgradeApp)
+		return trace.BadParameter(`There was an issue trying to verify %v cluster image integrity.
```

Comment:
`Some required dependencies are missing from the cluster store for image %v. Check logs for more details.` should suffice